crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2267,13 +2218,12 @@ impl<'db> Type<'db> {
     /// `receiver_ty` must be `Type::Instance(_)` or `Type::ClassLiteral`.
     ///
     /// TODO: handle `super()` objects properly
-    #[must_use]
     fn call_bound(
         self,
         db: &'db dyn Db,
         receiver_ty: &Type<'db>,
         arguments: &CallArguments<'_, 'db>,
-    ) -> CallOutcome<'db> {
+    ) -> Result<CallOutcome<'db>, CallError<'db>> {
```

Comment:
Is it worth aliasing this type as `CallResult`?

Reply:
I prefer repeating types because I find it easier to read the code unless the type is used in many places, which isn't the case here (I think it's only three places today and David's about to remove `call_bound`, leaving only 2o